wain-syntax-binary/src/parser.rs

```diff
@@ -927,9 +953,20 @@ impl<'s> Parse<'s> for Code {
         let mut inner = parser.sub_parser(size, "code section")?;
         parser.eat(size);
 
-        let mut locals = vec![];
+        let mut locals_check = vec![];
+        let mut count = 0u32;
         for loc in inner.parse_vec::<Locals>()? {
             let loc = loc?;
+            if let Some(c) = count.checked_add(loc.count) {
+                count = c;
+            } else {
+                return Err(parser.error(ErrorKind::TooManyLocalVariables));
+            }
+            locals_check.push(loc);
+        }
+
+        let mut locals = vec![];
```

Comment:
```suggestion
        let mut locals = Vec::with_capacity(count as usize);
```

個数が分かってるので，最初に一気に確保しちゃってください

Reply:
なるほど！